src/sentry/event_manager.py

```diff
@@ -1321,6 +1333,9 @@ def _handle_regression(self, group, event, release):
         elif GroupResolution.has_resolution(group, release):
             return
 
+        elif has_pending_commit_resolution(group):
+            return
```

Comment:
Do we need to be concerned about the scenario where someone `fixes proj-123` in a commit message and then never ships that commit? Our push webhooks gobble up every commit on every branch.

Reply:
I'd say yes and no. Yes that's a problem, but no I'm not sure we should have to do anything about it. Otherwise I'm not really sure how we would treat this case since ultimately I'd *expect* Sentry to tell me somethings resolved when I've told it as such. I'm open to ideas here though. Could be some kind of timeout on resolution? (if commit doesn't land in a release in X days you assume its never going to happen and let people know?)

Reply:
Another thought here: if you mark something as fixing an issue, i would definitely expect that commit to be released. So food for thought on showing "unreleased changes".

Reply:
A timeout could work well. As the issue will come back as a regression if it wasn't actually fixed in another way.

Reply:
There are 3571 (out of 6089 total) GroupLink's in production which match this query (thus seemingly orphaned):

```python
GroupLink.objects.filter(
    linked_type=GroupLink.LinkedType.commit,
    relationship=GroupLink.Relationship.resolves,
).extra(
    where=["NOT EXISTS(SELECT 1 FROM sentry_releasecommit where commit_id = sentry_grouplink.linked_id)"]
)
```

All of them are marked as unresolved in production. That's definitely a bit concerning, and I think requires some more investigation into whats going on. Should they be marked as resolved? Is this just people misconfiguring releases (probable).

Reply:
Moving comment into appropriate thread:

Ok this is less bad than I thought.

1. my query was wrong
2. adding in last_seen severely reduces results

there are 270 orphaned issues that are still happening anad have a resolution associated:

```
GroupLink.objects.filter(
    linked_type=GroupLink.LinkedType.commit,
    relationship=GroupLink.Relationship.resolves,
).extra(
    where=[
        "NOT EXISTS(SELECT 1 FROM sentry_releasecommit where commit_id = sentry_grouplink.linked_id)",
        "EXISTS(SELECT 1 FROM sentry_groupedmessage where id = sentry_grouplink.group_id AND status = 0 AND last_seen > now() - interval '7 days')",
    ]
)
```

For clarity this query finds Group Links which:

1. are not released yet
2. the issue is not resolved
3. the issue has happened in the last 7 days

Reply:
Repository deletion does not remove GroupLink records that were connected to commits. And `cleanup.py` does not remove GroupLink records, nor is there a foreign key to Group from GroupLink. This could explain one source of orphans.